app/assets/javascripts/analytics/page-content.js

```diff
@@ -0,0 +1,120 @@
+(function () {
+  "use strict";
+  window.GOVUK = window.GOVUK || {};
+  var PageContent = function () { }
+
+  PageContent.getNumberOfSections = function () {
+    switch(true) {
+      case isNavigationGridPage():
+        return $('a[data-track-category="navGridLinkClicked"]').length;
+      case isNavigationAccordionPage():
+        return $('[data-track-count="accordionSection"]').length;
+      case isDocumentCollectionPage():
+        return $('.document-collection .group-title').length;
+      case isMainstreamBrowsePage():
+        return $('#subsection ul:visible').length || $('#section ul').length;
+      case isTopicPage():
+        return $('.topics-page nav.index-list').length;
+      case isPolicyAreaPage():
+        return $('.topic section h1.label').length;
+      default:
+        // It's a content page, not a "finding" page
+        var sidebarSections = $('[data-track-count="sidebarRelatedItemSection"]').length;
+        var sidebarTaxons = $('[data-track-count="sidebarTaxonSection"]').length;
+
+        return sidebarSections || sidebarTaxons;
+    }
+  };
+
+  PageContent.getNumberOfLinks = function () {
+    switch(true) {
+      case isNavigationGridPage():
```

Comment:
Good question. Before, the order was important because of document collections. They are both navigation pages and content pages, so the return statement needed a specific order. But with this `switch` statement, we look for navigation pages first (and they are findable in a deterministic way now since https://github.com/alphagov/whitehall/pull/3358), and default to a content page at the bottom. That means we don't rely on ordering anymore.